### PR TITLE
KOGITO-3994 Use Qute templating to handle Data Index config

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@nmirasch @pefernan take a look if you agree with this new approach if so, I'll apply it to the consoles.

**Review by nmirasch — APPROVED** [review]
@cristianonicolai long time looking for this connection and now quarkus is doing that for us. It looks very clean. I think we need to apply this at the consoles for sure.

File: kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java [hunk 1]
```diff
@@ -15,57 +15,29 @@
...
-            resource = Buffer.buffer(vertx.fileSystem()
-                                             .readFileBlocking("META-INF/resources/index.html")
-                                             .toString(UTF_8)
-                                             .replace("__GRAPHQL_HTTP_ENDPOINT__", "\"" + dataIndexHttpURL + "/graphql\"")
-                                             .replace("__GRAPHQL_WS_ENDPOINT__", "\"" + dataIndexWsURL + "/graphql\""));
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
+    Template index;
```
Comment (radtriste) [inline kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: maybe rename to `indexTpl`

File: kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java [hunk 2]
```diff
@@ -15,57 +15,29 @@
...
 
     void setupRouter(@Observes Router router) {
         router.route().handler(LoggerHandler.create());
         router.route().handler(FaviconHandler.create());
         router.route().handler(StaticHandler.create());
-        router.route(GET, "/").handler(ctx -> ctx.response().end(resource));
+        String template = index.render();
```
Comment (radtriste) [inline kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: maybe rename `template` -> `indexPage`

**Review by radtriste — COMMENTED** [review]
nitpicking

File: kogito-travel-agency/extended/travels/src/main/resources/templates/index.html [hunk 3]
```diff
@@ -55,7 +71,7 @@ <h1>Travels</h1>
 </div>
 </div>
 <script>
-    var graphql = __GRAPHQL_HTTP_ENDPOINT__;
+    var graphql = '{config:property('kogito.dataindex.http.url') or 'http://localhost:8180'}/graphql';
```
Comment (pefernan) [inline kogito-travel-agency/extended/travels/src/main/resources/templates/index.html]: Nitpicking too.. but could we have graphql as const?

File: kogito-travel-agency/extended/visas/src/main/resources/templates/index.html [hunk 4]
```diff
@@ -48,7 +64,7 @@ <h1>Visa applications</h1>
 </div>
 </div>
 <script>
-    var graphql = __GRAPHQL_HTTP_ENDPOINT__;
+    var graphql = '{config:property('kogito.dataindex.http.url') or 'http://localhost:8180'}/graphql';
```
Comment (pefernan) [inline kogito-travel-agency/extended/visas/src/main/resources/templates/index.html]: Same

**Review by pefernan — APPROVED** [review]
Good point! I like this approach better than the one we have on the consoles. Just dropped some comments but I'm OK with the code as it is.

File: kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java (hunk 1 above)
Comment (cristianonicolai) [reply kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: this matches with the file name that gets injected, so not sure, wdyt?

Comment (radtriste) [reply kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: To know it is a template and not the index page rendered

Comment (cristianonicolai) [reply kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: done

File: kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java (hunk 2 above)
Comment (cristianonicolai) [reply kogito-travel-agency/extended/travels/src/main/java/org/kie/kogito/app/VertxRouter.java]: done

File: kogito-travel-agency/extended/travels/src/main/resources/templates/index.html (hunk 3 above)
Comment (cristianonicolai) [reply kogito-travel-agency/extended/travels/src/main/resources/templates/index.html]: done

File: kogito-travel-agency/extended/visas/src/main/resources/templates/index.html (hunk 4 above)
Comment (cristianonicolai) [reply kogito-travel-agency/extended/visas/src/main/resources/templates/index.html]: done

**Review by radtriste — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**